src/ft/mod.rs

```diff
@@ -470,25 +471,30 @@ impl FreeTypeRasterizer {
 
         glyph.render_glyph(face.render_mode)?;
 
-        let (pixel_height, pixel_width, buf) = Self::normalize_buffer(&glyph.bitmap(), &face.rgba)?;
+        let (pixel_height, pixel_width, buffer) =
+            Self::normalize_buffer(&glyph.bitmap(), &face.rgba)?;
 
         let rasterized_glyph = RasterizedGlyph {
-            c: glyph_key.c,
+            character: glyph_key.character,
             top: glyph.bitmap_top(),
             left: glyph.bitmap_left(),
             width: pixel_width,
             height: pixel_height,
-            buf,
+            buffer,
         };
 
-        if face.colored {
-            let fixup_factor = if let Some(pixelsize_fixup_factor) = face.pixelsize_fixup_factor {
-                pixelsize_fixup_factor
-            } else {
-                // Fallback if user has bitmap scaling disabled.
-                let metrics = face.ft_face.size_metrics().ok_or(Error::MissingSizeMetrics)?;
-                f64::from(pixelsize) / f64::from(metrics.y_ppem)
+        if is_missing_glyph {
+            Err(Error::MissingGlyph(rasterized_glyph))
+        } else if face.colored {
```

Comment:
What do you think about extracting the `face.colored` check? Fundamentally it just mutates the glyph and has nothing to do with returning something different.

I was thinking something like this maybe:

```rust
let mut rasterized_glyph = ...;

if face.colored {
    ...

    downsample_bitmap(&mut rasterized_glyph, fixup_factor);
}

if glyph_index == MISSING_GLYPH_INDEX {
    Err(Error::MissingGlyph(rasterized_glyph))
} else {
    Ok(rasterized_glyph)
}
```